### PR TITLE
CppCheck PlantPipingSystemsManager

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Fixes programming errors
 - Use CppCheck as guide

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [ ] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/PlantPipingSystemsManager.cc [hunk 1]
```diff
@@ -1668,9 +1664,6 @@ namespace PlantPipingSystemsManager {
                                                                             DataLoopNode::ConnectionType::Outlet,
                                                                             NodeInputManager::CompFluidStream::Primary,
                                                                             DataLoopNode::ObjectIsNotParent);
-            if (thisCircuit.OutletNodeNum == 0) {
-                CurIndex = 3;
-            }
```
Comment (rraustad) [inline src/EnergyPlus/PlantPipingSystemsManager.cc:1673]: [src/EnergyPlus/PlantPipingSystemsManager.cc:1659]:(style),[unreadVariable],Variable 'CurIndex' is assigned a value that is never used.
[src/EnergyPlus/PlantPipingSystemsManager.cc:1672]:(style),[unreadVariable],Variable 'CurIndex' is assigned a value that is never used.

File: src/EnergyPlus/PlantPipingSystemsManager.cc [hunk 2]
```diff
@@ -3142,13 +3135,13 @@ namespace PlantPipingSystemsManager {
...
         //       RE-ENGINEERED  na
 
-        int cellCountUpToNow = 0;
         std::vector<Real64> tempCellWidths;
 
         if (PartitionsExist) {
 
             for (int i = 0; i < (int)ThesePartitionRegions.size(); ++i) {
                 auto &thisPartition = ThesePartitionRegions[i];
+                int cellCountUpToNow = 0;
```
Comment (rraustad) [inline src/EnergyPlus/PlantPipingSystemsManager.cc]: [src/EnergyPlus/PlantPipingSystemsManager.cc:3145]:(style),[variableScope],The scope of the variable 'cellCountUpToNow' can be reduced.

File: src/EnergyPlus/PlantPipingSystemsManager.cc [hunk 3]
```diff
@@ -3459,22 +3452,20 @@ namespace PlantPipingSystemsManager {
...
                                             cellType = CellType::VertInsulation;
                                             ++NumInsulationCells;
                                         }
                                     } else { // Vertical insulation extends to depth of basement floor
-                                        if (CellYIndex <= this->y_max_index && CellYIndex > YFloorIndex) {
-                                            cellType = CellType::VertInsulation;
-                                            ++NumInsulationCells;
-                                        }
+                                        cellType = CellType::VertInsulation;
+                                        ++NumInsulationCells;
```
Comment (rraustad) [inline src/EnergyPlus/PlantPipingSystemsManager.cc:3461]: [src/EnergyPlus/PlantPipingSystemsManager.cc:3462]:(style),[knownConditionTrueFalse],Condition 'CellYIndex<=this->y_max_index' is always true
[src/EnergyPlus/PlantPipingSystemsManager.cc:3467]:(style),[knownConditionTrueFalse],Condition 'CellYIndex<=this->y_max_index' is always true
[src/EnergyPlus/PlantPipingSystemsManager.cc:3467]:(style),[knownConditionTrueFalse],Condition 'CellYIndex>YFloorIndex' is always true

CellYIndex > YFloorIndex is always true (line 3458/3451)

File: src/EnergyPlus/PlantPipingSystemsManager.cc [hunk 4]
```diff
@@ -3459,22 +3452,20 @@ namespace PlantPipingSystemsManager {
...
                                     }
                                 }
                             }
                         } else if (CellYIndex == this->y_max_index) { // Surface cells
                             cellType = CellType::GroundSurface;
                             ++NumGroundSurfaceCells;
-                        } else if (CellYIndex == 0 || CellXIndex == 0 || CellZIndex == 0) { // Farfield boundary
+                        } else if (CellXIndex == 0 || CellZIndex == 0) { // Farfield boundary
```
Comment (rraustad) [inline src/EnergyPlus/PlantPipingSystemsManager.cc:3468]: [src/EnergyPlus/PlantPipingSystemsManager.cc:3477]:(style),[knownConditionTrueFalse],Condition 'CellYIndex==0' is always false

See line 3429/3422

File: src/EnergyPlus/PlantPipingSystemsManager.cc [hunk 5]
```diff
@@ -5858,12 +5849,12 @@ namespace PlantPipingSystemsManager {
...
             rhoCp = this->Moisture.rhoCP_soil_ice;
-        } else if ((CellTemp > frzAllIce) && (CellTemp < frzIceTrans)) { // in between totally frozen and ice transition
+        } else if (CellTemp < frzIceTrans) { // in between totally frozen and ice transition
             rhoCp = this->Moisture.rhoCP_soil_ice +
                     (this->Moisture.rhoCP_soil_transient - this->Moisture.rhoCP_soil_ice) / (frzIceTrans - frzAllIce) * (CellTemp - frzAllIce);
-        } else if ((CellTemp >= frzIceTrans) && (CellTemp <= frzLiqTrans)) { // in between ice transition and liquid transition
+        } else if (CellTemp <= frzLiqTrans) { // in between ice transition and liquid transition
             rhoCp = this->Moisture.rhoCP_soil_transient;
-        } else if ((CellTemp > frzLiqTrans) && (CellTemp < frzAllLiq)) { // in between liquid transition and all liquid
+        } else if (CellTemp < frzAllLiq) { // in between liquid transition and all liquid
```
Comment (rraustad) [inline src/EnergyPlus/PlantPipingSystemsManager.cc:5857]: [src/EnergyPlus/PlantPipingSystemsManager.cc:5861]:(style),[knownConditionTrueFalse],Condition 'CellTemp>frzAllIce' is always true

And then the other 2 suggestions popped up after this was fixed.

File: src/EnergyPlus/PlantPressureSystem.cc [hunk 6]
```diff
@@ -493,25 +485,27 @@ void UpdatePressureDrop(EnergyPlusData &state, int const LoopNum)
...
 
         // Split here based on a single branch loop or a splitter/mixer configuration
         if (NumBranches == 1) { // Just do the single branch
 
             //***SINGLE BRANCH***!
             BranchNum = 1;
-            DistributePressureOnBranch(state, LoopNum, LoopSideNum, BranchNum, BranchPressureDrop, FoundAPumpOnBranch);
-            LoopSidePressureDrop += BranchPressureDrop;
-            LoopPressureDrop += BranchPressureDrop;
+            Real64 BranchPressureDropValue = 0.0;
```
Comment (rraustad) [inline src/EnergyPlus/PlantPressureSystem.cc:495]: BranchPressureDrop is also a function name which showed as a shadow outer function error after this scope change so I changed the variable name.

void BranchPressureDrop(EnergyPlusData &state,

File: src/EnergyPlus/PlantPressureSystem.cc [hunk 7]
```diff
@@ -753,7 +743,7 @@ void PassPressureAcrossMixer(
 void PassPressureAcrossSplitter(EnergyPlusData &state,
                                 int const LoopNum,
                                 const DataPlant::LoopSideLocation LoopSideNum,
-                                Real64 &SplitterInletPressure)
+                                Real64 const SplitterInletPressure)
```
Comment (rraustad) [inline src/EnergyPlus/PlantPressureSystem.cc:746]: Also revert reference to scalar.

[src/EnergyPlus/PlantPressureSystem.cc:728]:(style),[constParameter],Parameter 'MixerPressure' can be declared as reference to const
[src/EnergyPlus/PlantPressureSystem.cc:756]:(style),[constParameter],Parameter 'SplitterInletPressure' can be declared as reference to const

**Comment by rraustad** [comment]
I've been lucky with diffs up until now. I wonder which change caused this?

**Comment by Myoldmopar** [comment]
Good catch, all clean again!  And happy with develop locally, this is going in now, thanks for the flurry of cppcheck fixes!

**Pull request merged**